src/terrain/tile.cpp

```diff
@@ -9,88 +9,93 @@
 namespace terrain {
 
 Tile::Tile() {
-    init({1, 1, 1}, false);
+    init({1, 1, 1}, 0);
 }
 
-Tile::Tile(std::array<int, 3> sop, const Material* mat) {
-    init(sop, mat);
+Tile::Tile(std::array<int, 3> sop, const terrain::Material* material) {
+    init(sop, material);
+    set_material(material, 0);
 }
 
-Tile::Tile(std::array<int, 3> sop, const Material* mat_, uint8_t color_id_) {
-    init(sop, mat_, color_id_);
+Tile::Tile(
+    std::array<int, 3> sop, const terrain::Material* material, uint8_t color_id
+) {
+    // bool solid = materials[mat_id].solid;
+    init(sop, material);
+    set_material(material, color_id);
 }
 
 void
-Tile::init(std::array<int, 3> sop, bool solid_) {
+Tile::init(std::array<int, 3> sop, const terrain::Material* mat) {
```

Comment:
This should really be an initializer list

Reply:
this is not a constructor.

Reply:
Well why isn't it a constructor?

Reply:
well tiles are defult initialized, then initializd a second time with init.
I could do the second initialization with set material